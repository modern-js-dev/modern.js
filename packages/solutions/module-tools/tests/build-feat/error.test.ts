import {
  padSpaceWith,
  InternalBuildError,
} from '../../src/features/build/error';

describe('padSpaceWith', () => {
  test(`str is 'hello', target length is 6, endStr is '.'`, () => {
    expect(padSpaceWith('hello', 6, { endStr: '.' })).toBe('hello.');
  });

  test(`str is 'hello', target length is 10, endStr is '.'`, () => {
    expect(padSpaceWith('hello', 10, { endStr: '.' })).toBe('hello    .');
  });

  test(`str is 'hello', target length is 5, endStr is '.'`, () => {
    expect(padSpaceWith('hello', 5, { endStr: '.' })).toBe('hello');
  });

  test(`str is 'hello', target length is 5, endStr is ''`, () => {
    expect(padSpaceWith('hello', 5, { endStr: '' })).toBe('hello');
  });

  test(`str is 'hello', target length is 10, endStr is '.', style is s => s + ' world'`, () => {
    expect(
      padSpaceWith('hello', 6, { endStr: '.', style: s => `${s} world` }),
    ).toBe('hello world.');
  });
});

describe('InternalBuildError', () => {
  test('buildType is bundle', () => {
    const e = new InternalBuildError(new Error('this is error'), {
      buildType: 'bundle',
      format: 'cjs',
      target: 'es2015',
    });
    expect(e.formatError()).toMatchSnapshot();
  });

  test('buildType is bundleless', () => {
    const e = new InternalBuildError(new Error('this is error'), {
      buildType: 'bundleless',
      format: 'cjs',
      target: 'es2015',
    });
    expect(e.formatError()).toMatchSnapshot();
  });
});
