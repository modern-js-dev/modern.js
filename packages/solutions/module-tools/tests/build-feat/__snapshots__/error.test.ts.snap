// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalBuildError buildType is bundle 1`] = `
Array [
  "╭───────────────────────╮",
  "│ bundle failed:        │",
  "│  - format is \\"cjs\\"    │",
  "│  - target is \\"es2015\\" │",
  "╰───────────────────────╯",
  "
Detailed Information: ",
  "Error: this is error",
  "",
]
`;

exports[`InternalBuildError buildType is bundleless 1`] = `
Array [
  "╭───────────────────────╮",
  "│ bundleless failed:    │",
  "│  - format is \\"cjs\\"    │",
  "│  - target is \\"es2015\\" │",
  "╰───────────────────────╯",
  "
Detailed Information: ",
  "Error: this is error",
  "",
]
`;
